python/mage/link_prediction/link_prediction_util.py

```diff
@@ -0,0 +1,382 @@
+    graph_edges_len = len(edge_type_u)
+    eids = np.arange(graph_edges_len)  # get all edge ids from number of edges and create a numpy vector from it.
+    eids = np.random.permutation(eids)  # randomly permute edges
+
+    # Get training and validation edges
+    tr_eids, val_eids = eids[val_size:], eids[:val_size]
+
+    # Create and masks that will be used in the batch training
+    train_eid_dict, val_eid_dict = {target_relation: tr_eids}, {target_relation: val_eids}
+
+    return train_eid_dict, val_eid_dict
+
+def classify(probs: torch.tensor, threshold: float) -> torch.tensor:
+    """Classifies based on probabilities of the class with the label one.
+
+    Args:
+        probs (torch.tensor): Edge probabilities.
+
+    Returns:
+        torch.tensor: classes
+    """
+
+    return probs > threshold
+
+def evaluate(metrics: List[str], labels: torch.tensor, probs: torch.tensor, result: Dict[str, float], threshold: float, epoch: int, loss: float,
+            operator: Callable[[float, float], float]) -> None:
+    """Returns all metrics specified in metrics list based on labels and predicted classes. In-place modification of dictionary.
+
+    Args:
+        metrics (List[str]): List of string metrics.
+        labels (torch.tensor): Predefined labels.
+        probs (torch.tensor): Probabilities of the class with the label one.
+        result (Dict[str, float]): Prepopulated result that needs to be modified.
+        threshold (float): classification threshold. 0.5 for sigmoid etc.
+    Returns:
+        Dict[str, float]: Metrics embedded in dictionary -> name-value shape
+    """
+    classes = classify(probs, threshold)
+    result[EPOCH] = epoch
+    result[LOSS] = operator(result[LOSS], loss)
+    for metric_name in metrics:
+        if metric_name == ACCURACY:
+            result[ACCURACY] = operator(result[ACCURACY], accuracy_score(labels, classes))
+        elif metric_name == AUC_SCORE:
+            result[AUC_SCORE] = operator(result[AUC_SCORE], roc_auc_score(labels, probs.detach()))
+        elif metric_name == F1:
+            result[F1] = operator(result[F1], f1_score(labels, classes))
+        elif metric_name == PRECISION:
+            result[PRECISION] = operator(result[PRECISION], precision_score(labels, classes))
+        elif metric_name == RECALL:
+            result[RECALL] = operator(result[RECALL], recall_score(labels, classes))
+        elif metric_name == POS_PRED_EXAMPLES:
+            result[POS_PRED_EXAMPLES] = operator(result[POS_PRED_EXAMPLES], (probs > 0.5).sum().item())
+        elif metric_name == NEG_PRED_EXAMPLES:
+            result[NEG_PRED_EXAMPLES] = operator(result[NEG_PRED_EXAMPLES], (probs < 0.5).sum().item())
+        elif metric_name == POS_EXAMPLES:
+            result[POS_EXAMPLES] = operator(result[POS_EXAMPLES], (labels == 1).sum().item())
+        elif metric_name == NEG_EXAMPLES:
+            result[NEG_EXAMPLES] = operator(result[NEG_EXAMPLES], (labels == 0).sum().item())
```

Comment:
This might be disentangled

Reply:
Do you have some ideas? I thought that the idea of the lambda function(operator) was good.